cartography/intel/opal/__init__.py

```diff
@@ -0,0 +1,149 @@
+import logging
+from typing import Any
+from typing import Dict
+from typing import List
+
+import neo4j
+import opal_security as opal
+
+from cartography.config import Config
+from cartography.intel.opal.opal_aws import sync_opal_aws
+from cartography.intel.opal.opal_okta import sync_opal_okta
+from cartography.util import run_cleanup_job
+from cartography.util import timeit
+
+PAGE_SIZE = 200
+logger = logging.getLogger(__name__)
+
+
+def _resource_client(configuration: opal.Configuration) -> opal.api.ResourcesApi:
+    api_client = opal.ApiClient(configuration)
+    return opal.ResourcesApi(api_client)
+
+
+def _group_client(configuration: opal.Configuration) -> opal.api.GroupsApi:
+    api_client = opal.ApiClient(configuration)
+    return opal.GroupsApi(api_client)
+
+
+def _owners_client(configuration: opal.Configuration) -> opal.api.OwnersApi:
+    api_client = opal.ApiClient(configuration)
+    return opal.OwnersApi(api_client)
+
+
+def parse_group(group: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Parse a group dict and add the resource_type field.
+    """
+    group["resource_id"] = group.get("group_id")
+    group["resource_type"] = "OPAL_GROUP"
+    group["remote_resource_name"] = group.get("remote_name")
+    return group
+
+
+def get_all_groups(client: opal.api.GroupsApi) -> List[Dict[str, Any]]:
+    """
+    Retrieve all groups from the Opal API.
+
+    :param client: The Opal GroupsApi client.
+    :return: A list of all groups.
+    """
+    logger.info("Getting all groups from Opal")
+    try:
+        response = client.get_groups(page_size=PAGE_SIZE)
+        groups = [parse_group(g.to_dict()) for g in response.results]
+        while response.next:
+            response = client.get_groups(page_size=PAGE_SIZE, cursor=response.next)
+            groups.extend([parse_group(g.to_dict()) for g in response.results])
+    except opal.ApiException as e:
+        logger.error(f"Error getting groups from Opal: {e}")
+        raise
+    return groups
+
+
+def get_opal_groups_to_okta_groups_map(groups: List[Dict[str, Any]]) -> Dict[str, str]:
+    """
+    Create a map of Opal Groups to their Okta groups using the remote_info field.
+    """
+    logger.info("Mapping opal groups to okta groups")
+
+    opal_to_okta_map = {}
+    for group in groups:
+        opal_group_id = group.get('group_id')
+        okta_group_id = group.get('remote_id')
+        if opal_group_id and okta_group_id:
+            opal_to_okta_map[opal_group_id] = okta_group_id
+    return opal_to_okta_map
+
+
+def get_owner_ids_to_okta_groups_map(
```

Comment:
Would it make sense to move this and get_opal_groups_to_okta_groups_map to opal_okta.py?